zebra/rt_netlink.c

```diff
@@ -4226,16 +4235,39 @@ static int netlink_macfdb_change(struct nlmsghdr *h, int len, ns_id_t ns_id)
 		}
 
 		if (IS_ZEBRA_IF_VXLAN(ifp)) {
-			if (!dst_present)
+			if (zebra_mac_ext_learn_mode()) {
+				/* the Linux kernel should never give us an entry
+				 * with VxLAN info
+				 */
 				return 0;
+			} else {
+				if (!dst_present)
+					return 0;
 
-			if (vni_mcast_grp)
-				return zebra_vxlan_if_vni_mcast_group_add_update(
-					ifp, vni, &vtep_ip);
+				if (vni_mcast_grp)
+					return zebra_vxlan_if_vni_mcast_group_add_update(ifp, vni,
+											 &vtep_ip);
 
-			return zebra_vxlan_dp_network_mac_add(
-				ifp, br_if, &mac, vid, vni, nhg_id, sticky,
-				!!(ndm->ndm_flags & NTF_EXT_LEARNED));
+				return zebra_vxlan_dp_network_mac_add(ifp, br_if, &mac, vid, vni,
+								      nhg_id, sticky,
+								      !!(ndm->ndm_flags &
+									 NTF_EXT_LEARNED));
+			}
+		}
+
+		if (zebra_mac_ext_learn_mode()) {
+			if (!(ndm->ndm_flags & NTF_EXT_LEARNED)) {
+				if (IS_ZEBRA_DEBUG_KERNEL)
+					zlog_debug("        Should drop entry due to missing extern learn, FLAGS = 0x%x",
+						   ndm->ndm_flags & NTF_EXT_LEARNED);
```

Comment:
If we want to print all flags `FLAGS = `, then why we expect only `NTF_EXT_LEARNED`?